ui/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/TagForm/TagChip/TagChip.tsx

```diff
@@ -0,0 +1,33 @@
+import type { ChipProps, PropsWithSpread } from "@canonical/react-components";
+import { Chip } from "@canonical/react-components";
+
+import type { TagIdCountMap } from "app/store/machine/utils";
+import type { Tag } from "app/store/tag/types";
+
+type Props = PropsWithSpread<
+  {
+    machineCount: number;
+    tag: Tag;
+    tagIdsAndCounts: TagIdCountMap;
+  },
+  Partial<ChipProps>
+>;
+
+export const TagChip = ({
+  machineCount,
+  tag,
+  tagIdsAndCounts,
+  ...props
+}: Props): JSX.Element => {
+  const tagCount = tagIdsAndCounts.get(tag.id);
+  return (
+    <Chip
+      className="is-inline"
```

Comment:
It'd be good if the text inside the chips wrapped instead of truncated - I think it's not very useful otherwise

Reply:
It looks like Vanilla doesn't provide a way to not have truncated text. I've asked a question on the design and will followup if needed:

https://www.figma.com/file/AFw4WN623gie2Fi34btFlZ?node-id=1329:20859#171562651